cocotb/__init__.py

```diff
@@ -192,6 +192,30 @@ def _initialise_testbench(argv_):
     if not sys.warnoptions:
         warnings.simplefilter("default")
 
+    attach_str = os.getenv("COCOTB_PY_ATTACH")
+    if attach_str is not None:
+
+        try:
+            if ":" not in attach_str:
+                raise ValueError("Invalid COCOTB_PY_ATTACH, form is 'host:port'.")
+            host_str, port_str = attach_str.split(":", 1)
+            port_num = int(port_str)
+        except Exception as e:
+            raise RuntimeError("Failure to parse COCOTB_PY_ATTACH ('{}')".format(attach_str)) from e
+
+        try:
+            import debugpy
+        except ImportError:
+            cocotb.log.error("COCOTB_PY_ATTACH is set but debugpy package is not importable. Install debugpy to debug cocotb.")
+        
+        try:
+            listen_host, listen_info = debugpy.listen((host_str, port_num))
+            cocotb.log.info("Waiting for Python debugger attach on {}:{}".format(listen_host, listen_info))
+            debugpy.wait_for_client()
+            debugpy.breakpoint()
```

Comment:
Not sure if it's better to break into the debugger here or rely solely on user breakpoints.